### DISCUSSION TITLE
Making bot persistent on cloud

### DISCUSSION THREAD

**Discussion by arpitgoyall** [python-telegram-bot/python-telegram-bot#2668] [Q&A] [answered]
Hey i am making a telegram bot which saves data of user through conversation handler. This is Minimal work example https://pastebin.com/HWCtP5LW I want to deploy it on heroku and setting its database url in This [ptcontrib postgres constructor](https://github.com/python-telegram-bot/ptbcontrib/tree/main/ptbcontrib/postgres_persistence)

The problem i am facing is that whenever i restart the script or bot gets turned off, data is not persist, we have to again put the data which is not good, so please tell me what i am doing wrong or what is missing?

**Comment by starry-shivam** [comment, marked as answer]
Hi this is fixed now, see https://github.com/python-telegram-bot/ptbcontrib/pull/32
